src/saturn_engine/worker/topic.py

```diff
@@ -66,7 +67,14 @@ async def run(self) -> AsyncGenerator[TopicOutput, None]:
                 yield message
 
             if len(messages) == 0:
-                await asyncio.sleep(self.sleep_time.total_seconds())
+                await self.wait()
+
+    async def wait(self) -> None:
+        await self.event.wait()
```

Comment:
you want a timeout, in most case it won't be `wake`n.

```
contextlib.suppress(TimeoutError):
  await asyncio.wait_for(self.event.wait(), timeout=self.sleep_time.total_seconds())
```

Reply:
ok thanks